### PR TITLE
PKG-4723 20240116.2

### PR DESCRIPTION
**Pull request by cbouss**
abseil-cpp 20240116.2

**Destination channel:** main

### Links

- https://anaconda.atlassian.net/browse/PKG-4723
- https://github.com/abseil/abseil-cpp/tree/20240116.2
- https://github.com/abseil/abseil-cpp/releases/tag/20240116.2
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/gtest-feedstock/pull/6 (string view support)

### Explanation of changes:
- sync conda-forge changes
  - https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075
  - https://github.com/conda-forge/abseil-cpp-feedstock/issues/45 
- adapt recipe
- tighten run_exports (patch release includes breaking changes)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,82 +1,200 @@
...
+        - Library/lib/cmake/absl/  # [win]
+
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja-base
+      host:
+        - gtest 1.14
```
Comment (danpetry) [inline recipe/meta.yaml:153]: I can't see where gtest is being used, or the test binaries are being used (beyond checking that they're present)?

**Review by danpetry — COMMENTED** [review]
Started looking at this, will split over multiple days as it's v big.
If you're able to isolate the relevant discussions for the changes on conda-forge that would be helpful; the links include discussions around historical updates, protobuf, etc. (or do I need to read the whole thing for context..?)

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml:153]: It's for abseil_test_dll, included in this pacakge.

**Comment by cbouss** [comment]
> Started looking at this, will split over multiple days as it's v big. If you're able to isolate the relevant discussions for the changes on conda-forge that would be helpful; the links include discussions around historical updates, protobuf, etc. (or do I need to read the whole thing for context..?)

Tp put it short, all packages depending on abseil must be built with cxx 17 at least. Also, even minor updates can induce breaking changes. The other relevant change is conda-forge renaming the package from abseil-cpp to libabseil.

File: recipe/meta.yaml (hunk 1 above)
Comment (danpetry) [reply recipe/meta.yaml:153]: do the test binaries get run? or does it ship a test suite for use by devs

Comment (cbouss) [reply recipe/meta.yaml:153]: libprotobuf does make use of abseil test binaries to built its own test binaries

**Review by danpetry — APPROVED** [review]

**Review by JamesRobertsonGames — APPROVED** [review]

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**